libs/blocks/youtube/youtube.js

```diff
@@ -14,12 +14,19 @@ export default function init(a) {
         webkitallowfullscreen mozallowfullscreen allowfullscreen
         allow="encrypted-media; accelerometer; gyroscope; picture-in-picture"
         scrolling="no"
+        id='player-${id}'
         title="${title}">
       </iframe>
     </div>`;
     a.insertAdjacentHTML('afterend', embedHTML);
     a.remove();
   };
-
   createIntersectionObserver({ el: a, callback: embedVideo });
+  document.onreadystatechange = () => {
```

Comment:
Couldn't a simple event listener be used instead of overhauling the functionality altogether? Like `document.addEventListener('readystatechange')`. Otherwise any other pieces of logic depending on this event might get impacted.

I also agree with @mokimo that we wouldn't want this tracked if the video is not actuallt embedded.

Reply:
done